native/game_backend/src/player.rs

```diff
@@ -342,6 +363,20 @@ impl Player {
             .iter()
             .any(|action_tracker| action_tracker.action != Action::Moving)
     }
+
+    pub fn can_perform_attack(&self, skill_key: &String) -> bool {
+        // Check if player is still performing an action or if skill is still on cooldown.
+        self.can_do_action()
+            && (can_perform_basic_attack(self) || can_perform_skill_attack(self, skill_key))
+    }
+}
+
+fn can_perform_basic_attack(player: &Player) -> bool {
+    player.cooldowns.contains_key(BASIC_SKILL_KEY) && player.available_burst_loads > 0
```

Comment:
This is not working as expected, since we need to be able to perform a basic attack to this clause to be true `player.cooldowns.contains_key(BASIC_SKILL_KEY)` Replacing this with some check like the one from the previous comment would do